### PR TITLE
# Only email application owners if they have email addresses for cert…

### PR DESCRIPTION
**Pull request by kenny-lee-1**
…ain notifications"

### CODE REVIEW

File: app/models/notification.rb [hunk 1]
```diff
@@ -41,4 +41,17 @@ def send_admin_email
...
+  # attempt to not tread on MBIS application behaviour for now until
+  # notifications ana mail is fully decoupled
+  def project_user_ids(project_id)
+    return if project_id.nil?
+
+    application = Project.find(project_id)
+
+    users_associated_to_application = Project.find(project_id).users
+    users_associated_to_application.each_with_object([]) do |user, ids|
+      ids << user.id if user.email.match?(/@phe.gov.uk\z/)
```
Comment (miles-smith) [inline app/models/notification.rb:54]: `user.internal?`

File: app/models/notification.rb [hunk 2]
```diff
@@ -41,4 +41,17 @@ def send_admin_email
     return unless admin_users
     NotificationMailer.send_admin_messages(self).deliver_later
   end
+
+  # attempt to not tread on MBIS application behaviour for now until
+  # notifications ana mail is fully decoupled
+  def project_user_ids(project_id)
+    return if project_id.nil?
+
+    application = Project.find(project_id)
```
Comment (miles-smith) [inline app/models/notification.rb:50]: Worth setting up an association and doing this in the Rails-y way?

File: app/models/notification.rb (hunk 1 above)
Comment (kenny-lee-1) [reply app/models/notification.rb:54]: 👍

File: app/models/notification.rb [hunk 3]
```diff
@@ -41,4 +41,17 @@ def send_admin_email
...
+
+    users_associated_to_application = Project.find(project_id).users
```
Comment (miles-smith) [inline app/models/notification.rb:52]: As above, with associations in place we could walk them e.g. `project.users`, which then gives access to scopes for filtering and `:pluck` - we could boil this down to something along the lines of `project.users.internal.pluck(:id)`.

I know we're not dealing with huge collections of users so it's not like we'll be hitting performance bottlenecks, so I feel like I'm splitting hairs a little.

Comment (kenny-lee-1) [reply app/models/notification.rb:52]: Yup - we don't have an internal scope but I'll add one. In the back of my mind this did make it PHE domain specific so maybe this scope would have to eventually be configurable

**Review by miles-smith — COMMENTED** [review]
What happens when the project users are PHE users and not external applicants?

**Comment by kenny-lee-1** [comment]
> What happens when the project users are PHE users and not external applicants?

Informally been asked to not change that behaviour i.e they get the emails. I think I'll need to attempt the whole flow on beta and see if we have any more gaps - combining this with the stack branch that's been merged yesterday.

**Review by miles-smith — APPROVED** [review]
LGTM

**Pull request merged**